### PR TITLE
[READY] New Singularity Hammer texture.

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request
Adds a sprite for the singularity hammer, made by @Nebulacrity. (inhands when sprite jannie)
We're still missing inhands, but this lets us move the weird black effect to only when it's held in hand, which makes things nicer.

Have a gander ss13SingularityHammer

I've also done some cleanup on the processing so it's not always running, minor things.

## Why It's Good For The Game

image

## Changelog
:cl: Nebulacrity for the sprites, LemonInTheDark for the mule work 
add: Improved on the singularity hammer, drop it on the ground and mire the way she shines.
/:cl:

### CODE REVIEW

**Comment by Nebulacrity** [comment]
Oh hey, I did a thing for once.

**Comment by LemonInTheDark** [comment]
The sprite jannie commands

**Comment by PKPenguin321** [comment]
instead of "outhand" you usually just call it the "icon" though i have seen it called the "on-ground" in the past for clarity

File: code/game/objects/items/singularityhammer.dm [hunk 1]
```diff
@@ -80,6 +89,7 @@
 			playsound(user, 'sound/weapons/marauder.ogg', 50, TRUE)
 			var/turf/target = get_turf(A)
 			vortex(target,user)
+			START_PROCESSING(SSobj, src)
```
Comment (AnturK) [inline code/game/objects/items/singularityhammer.dm:92]: Use a timer instead of processing this. It's only used to recharge.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/game/objects/items/singularityhammer.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/game/objects/items/singularityhammer.dm]: Aight, that is kinda jank

**Comment by LemonInTheDark** [comment]
Less memey title, at nebula's request

**Comment by Shadowflame909** [comment]
Thank you for this, It looks really good.

File: code/game/objects/items/singularityhammer.dm [hunk 2]
```diff
@@ -35,16 +33,19 @@
...
+
+/obj/item/singularityhammer/dropped(mob/user, silent)
+	. = ..()
+	color = null
 
-/obj/item/singularityhammer/Destroy()
-	STOP_PROCESSING(SSobj, src)
-	return ..()
+/obj/item/singularityhammer/update_icon_state()
+	icon_state = "singularity_hammer0"
```
Comment (AnturK) [inline code/game/objects/items/singularityhammer.dm:45]: Needs . = ..() call.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/game/objects/items/singularityhammer.dm [hunk 3]
```diff
@@ -35,16 +33,19 @@
...
 
-/obj/item/singularityhammer/update_icon_state()
-	icon_state = "mjollnir0"
+/obj/item/singularityhammer/equipped(mob/user, slot, initial)
+	. = ..()
+	color = "#212121"
+
+/obj/item/singularityhammer/dropped(mob/user, silent)
+	. = ..()
+	color = null
```
Comment (AnturK) [inline code/game/objects/items/singularityhammer.dm:42]: Also use add/remove_atom_color wrapper here.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/game/objects/items/singularityhammer.dm (hunk 3 above)
Comment (LemonInTheDark) [reply code/game/objects/items/singularityhammer.dm:42]: The fact that add and remove color_atom swap argument order is really really sad.

**Comment by LemonInTheDark** [comment]
Done and dusted

File: code/game/objects/items/singularityhammer.dm [hunk 4]
```diff
@@ -35,16 +33,21 @@
 /obj/item/singularityhammer/proc/on_unwield(obj/item/source, mob/user)
 	wielded = FALSE
 
-/obj/item/singularityhammer/update_icon_state()
-	icon_state = "mjollnir0"
+/obj/item/singularityhammer/equipped(mob/user, slot, initial)
+	. = ..()
+	color = "#212121"
```
Comment (AnturK) [inline code/game/objects/items/singularityhammer.dm:38]: Forgot to remove this line.

Comment (LemonInTheDark) [reply code/game/objects/items/singularityhammer.dm]: ahhhhhh christ

File: code/game/objects/items/singularityhammer.dm [hunk 5]
```diff
@@ -72,14 +76,15 @@
 	if(!proximity)
 		return
 	if(wielded)
-		if(charged == 5)
-			charged = 0
+		if(charged == TRUE)
```
Comment (AnturK) [inline code/game/objects/items/singularityhammer.dm:79]: if(charged)

File: code/game/objects/items/singularityhammer.dm [hunk 6]
```diff
@@ -72,14 +76,15 @@
...
-			charged = 0
+		if(charged == TRUE)
+			charged = FALSE
 			if(istype(A, /mob/living/))
 				var/mob/living/Z = A
 				Z.take_bodypart_damage(20,0)
 			playsound(user, 'sound/weapons/marauder.ogg', 50, TRUE)
 			var/turf/target = get_turf(A)
 			vortex(target,user)
+			addtimer(CALLBACK(src, /obj/item/singularityhammer/proc/recharge), 100)
```
Comment (AnturK) [inline code/game/objects/items/singularityhammer.dm:87]: We generally use the `.proc/recharge` format on the same type.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/game/objects/items/singularityhammer.dm (hunk 6 above)
Comment (LemonInTheDark) [reply code/game/objects/items/singularityhammer.dm]: Gottcha

**Pull request merged**